external_webpage/request_handler_utils.py

```diff
@@ -51,15 +56,73 @@ def get_summary_details_of_all_instruments(data):
     return inst_data
 
 
-def get_detailed_state_of_specific_instrument(instrument, data):
+def get_instrument_time_since_epoch(instrument_data):
+    """
+    Return the instrument time as seconds since epoch.
+
+    Raises: KeyError: instrument time cannot be parsed from instrument_data
+    Raises: ValueError: if instrument time has wrong time format
+    """
+
+    time_format = '%m/%d/%Y %H:%M:%S'
+    try:
+        inst_time_str = instrument_data['inst_pvs']['TIME_OF_DAY']['value']
+    except KeyError:
+        logger.error("Cannot parse TIME_OF_DAY from instrument data.")
+        raise
+
+    try:
+        inst_time_struct = time.strptime(inst_time_str, time_format)
+    except ValueError:
+        logger.error("Instrument time does not match format {}.".format(time_format))
+        raise
+
+    inst_time = time.mktime(inst_time_struct)
+
+    return inst_time
+
+
+def check_out_of_sync(instrument_data, time_shift_threshold, extract_time_from_instrument_func=get_instrument_time_since_epoch, current_time_func=time.time):
+    """
+    Update the instrument data with the time shift to the webserver.
+
+    :param instrument_data: The data dictionary of an instrument
+    :param time_shift_threshold: If the time shift is greater than this value the data is considered outdated
+
+    Raises: ValueError: if time_diff cannot calculated
+    Raises: TypeError: if time_diff cannot calculated
+    """
+    try:
+        inst_time = extract_time_from_instrument_func(instrument_data)
+        current_time = current_time_func()
+        time_diff = abs(current_time - inst_time)
```

Comment:
We should also round it to the nearest second to avoid output like the following:

![image](https://user-images.githubusercontent.com/24375813/46009838-98826d00-c0b8-11e8-9500-c03756748401.png)